lib/sshutils/sftp/local.go

```diff
@@ -36,91 +37,112 @@ func (l localFS) Type() string {
 	return "local"
 }
 
-func (l localFS) Stat(ctx context.Context, path string) (os.FileInfo, error) {
+func (l *localFS) Glob(ctx context.Context, pattern string) ([]string, error) {
 	if err := ctx.Err(); err != nil {
 		return nil, err
 	}
 
-	fi, err := os.Stat(path)
+	matches, err := filepath.Glob(pattern)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	return fi, nil
+	return matches, nil
 }
 
-func (l localFS) ReadDir(ctx context.Context, path string) ([]os.FileInfo, error) {
+func (l localFS) Stat(ctx context.Context, p string) (os.FileInfo, error) {
 	if err := ctx.Err(); err != nil {
 		return nil, err
 	}
 
-	// normally os.ReadDir would be used as it's potentially more efficient,
-	// but because we want os.FileInfos of every file this is easier
-	f, err := os.Open(path)
+	fi, err := os.Stat(p)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	defer f.Close()
 
-	fileInfos, err := f.Readdir(-1)
+	return fi, nil
+}
+
+func (l localFS) ReadDir(ctx context.Context, p string) ([]os.FileInfo, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+
+	entries, err := os.ReadDir(p)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, err
+	}
+	fileInfos := make([]fs.FileInfo, len(entries))
+	for i, entry := range entries {
+		info, err := entry.Info()
+		if err != nil {
+			return nil, err
+		}
+		// if the file is a symlink, return the info of the linked file
+		if info.Mode().Type()&os.ModeSymlink != 0 {
+			info, err = os.Stat(filepath.Join(p, info.Name()))
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		fileInfos[i] = info
 	}
 
 	return fileInfos, nil
 }
 
-func (l localFS) Open(ctx context.Context, path string) (fs.File, error) {
+func (l localFS) Open(ctx context.Context, p string) (fs.File, error) {
```

Comment:
TBH I would've preferred if we kept `path` instead of `p` here and everywhere below. I don't think you're importing `path` in this file?